src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -338,8 +377,16 @@ private void addOtherTab(List<String> otherFields) {
 
     private List<String> addRequiredTab(EntryType type) {
         List<String> requiredFields = type.getRequiredFieldsFlat();
+        // build a list of all required fields, where each required field is followed by the hidden counter part
```

Comment:
And if you extract the for loop, you could write something like that in the javadoc